client/lib/cart/store/cart-synchronizer.js

```diff
@@ -106,17 +107,27 @@ CartSynchronizer.prototype.update = function( changeFunction ) {
 		// with a consistent state when we apply our local changes. The strategy
 		// here is to queue the changes until the first request has completed, then
 		// proceed as normal.
-		this._enqueueChange( changeFunction );
+		this._enqueueChange( changeFunction, serverFlushCallback );
+		if ( ! this._activeRequest ) {
+			this.fetch();
+		}
+		debug( 'update: change enqueued' );
 		return;
 	}
 
+	debug( 'update: processing change' );
+	this._addFlushCallback( serverFlushCallback );
+	this._processChangeFunction( changeFunction );
+	this.emit( 'change' );
+};
+
+CartSynchronizer.prototype._processChangeFunction = function ( changeFunction ) {
 	if ( this._activeRequest && this._activeRequest.state !== 'canceled' ) {
 		this._activeRequest.state = 'canceled';
 	}
 
 	this._latestValue = changeFunction( this._latestValue );
 	this._performRequest( 'update', this._postToServer.bind( this ) );
-	this.emit( 'change' );
```

Comment:
Also, maybe I'm missing something, but how can I be sure that the emitted 'synced' event is corresponding with the same `wrappedFunction` I called `update()` with ? As far as I understand that is not guaranteed, because if a request is in progress, the `wrappedFunction` of `update()` will be enqueued and `synced` event will be emitted for the previous completion sync, then, the enqueued `wrappedFunction` will proceed.